src/clone.c

```diff
@@ -140,8 +222,13 @@ SYS_FUNC(clone)
 			printaddr(tcp->u_arg[ARG_CTID]);
 			tprint_arg_name_end();
 		}
+		if (show_namespace && tcp->u_rval >= 0) {
+			tcp->auxstr = get_namespace_auxstr((int)tcp->u_rval, flags, tcp);
+			if (tcp->auxstr)
```

Comment:
When we need to check whether the syscall returned an error, we use `syserror(tcp)`.
That is, if `tcp->u_rval >= 0` is a syscall error check, it should rather be `!syserror(tcp)`.

If we want to check whether some `kernel_long_t val` is a positive int, then we would have to check
`val == (int) val && (int) val > 0`.

Reply:
I think the most appropriate condition here would be `flags & nsflags && !syserror(tcp)`.

Reply:
fixed locally.